libs/index/odc/index/stac.py

```diff
@@ -45,7 +45,7 @@ def _get_stac_bands(item, default_grid='g10.0m'):
             continue
 
         transform = asset['proj:transform']
-        grid = f'g{transform[0]}m'
+        grid = f'res:g{transform[0]}m'
```

Comment:
What I meant on Slack was `f"{whatever_variable_name_you_use_for_resolution:g}` ,so in this case 
```python
grid = f'{transform[0]:g}m'
```
so rather than converting to a string in a default fashion convert with "human readable floating point that might be an int" policy.